### ISSUE TITLE
Tests should be run with caching turned off (and on)

### ISSUE THREAD

**Issue by smillst** [typetools/checker-framework#601]
All tests should be run with caching turned off to ensure the results are identical to the results when caching is turned on.  (Though the test will take longer without caching)

Turning off caching in the AnnotatedTypeFactory results in three test failures: 
- Value Checker: value/Repo.java
- Interning Checker: interning/CompileTimeConstants.java
- NullnessChecker: jtreg/rawtypes/RawTypeFail.java

**Comment by smillst** [comment]
plume-lib: still type checks with caching turned off.  (plumb-lib with caching took 2.591 hours, without caching took 3.426 hours)

Daikon: have not tried to turn off caching, but I suspect that issue #565 is a caching bug.

**Comment by smillst** [comment]
Issue #516 is a worked around the fact that raw types are not create the same way when caching is turned off.

**Comment by smillst** [comment]
Raw Types:
The code below does not issue an error when caching is turned off.
````
Generic<@NonNull Object> generic = new Generic();
````
Currently, the Checker Framework attempts to infer type arguments for raw types.  (This is [document in the manual](http://types.cs.washington.edu/checker-framework/current/checker-framework-manual.html#generics-raw-types)).  During data flow, the raw type inference fails and the result is cached.  During type checking, the inference succeeds and the type the `new Generic()` expression is:
````
Generic<@NonNull Object>
```` 
Therefore the assignment succeeds.
